pybaseball/statcast.py

```diff
@@ -1,175 +1,70 @@
-
-            if smaller_data_2 is not None and not smaller_data_2.empty:
-                dataframe_list.append(smaller_data_2)
-                if verbose:
-                    print(f"Completed sub-query from {end_date} to {end_date} ({len(smaller_data_2)} results)")
-                retry_success = True
-            else:
-                print(f"Query unsuccessful for data from {end_date} to {end_date}. Skipping these dates.")
-
-            # Reset counter
-            error_counter = 0
-            break
-
-    if dataframe_list:
-        return pd.concat(dataframe_list, axis=0)
-
-    return None
-
-
+@cache.df_cache(expires=365)
 def small_request(start_dt: date, end_dt: date, team: Optional[str] = None, verbose: bool = False) -> pd.DataFrame:
     data = statcast_ds.get_statcast_data_from_csv_url(
         _SC_SMALL_REQUEST.format(start_dt=str(start_dt), end_dt=str(end_dt), team=team if team else '')
     )
     if data is not None and not data.empty:
-        if len(data) == _MAX_SC_RESULTS:
-            if verbose:
-                print('Query completed, but got the maximum number of results. Breaking it down to ensure consistency.')
-            day_span = (end_dt - start_dt).days
-            sub_chunk_1 = small_request(start_dt, start_dt + timedelta(days=day_span // 2), team=team, verbose=verbose)
-            if verbose and sub_chunk_1 is not None and not sub_chunk_1.empty:
-                print(
-                    f"Completed sub-query from {start_dt} to {start_dt + timedelta(days = day_span // 2)} ({len(sub_chunk_1)} results)"
-                )
-            sub_chunk_2 = small_request(
-                start_dt + timedelta(days=day_span // 2 + 1),
-                end_dt,
-                team=team,
-                verbose=verbose
-            )
-            if verbose and sub_chunk_2 is not None and not sub_chunk_2.empty:
-                print(
-                    f"Completed sub-query from {start_dt + timedelta(days = day_span // 2 + 1)} to {end_dt} ({len(sub_chunk_2)} results)"
-                )
-            data = pd.concat([sub_chunk_1, sub_chunk_2], axis=0)
         data = data.sort_values(
             ['game_date', 'game_pk', 'at_bat_number', 'pitch_number'],
             ascending=False
         )
+        if verbose:
+            print(f"Completed sub-query from {start_dt} to {end_dt} ({len(data)} results)")
 
     return data
 
 
-def large_request(start_date: date, end_date: date, step: int, verbose: bool,
+_OVERSIZE_WARNING = '''
+That's a nice request you got there. It'd be a shame if something were to happen to it.
+We strongly recommend that you enable caching before running this. It's as simple as `pybaseball.cache.enable()`.
+Since the Statcast requests can take a *really* long time to run, if something were to happen, like: a disconnect;
+gremlins; computer repair by associates of Rudy Giuliani; electromagnetic interference from metal trash cans; etc.;
```

Comment:
The Rudy Giuliani comment got a laugh out of me, but after news stories today (Oct 21) that might be a name we want to stay away from...

Reply:
Would 'Rudolph W. Giuliani' be better?

Reply:
"Former Mayors of New York City"

Reply:
I like it as-is, praise be Borat